src/main/java/com/erigitic/jobs/TEAction.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * This file is part of Total Economy, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) Eric Grandt <https://www.ericgrandt.com>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.erigitic.jobs;
+
+import ninja.leaping.configurate.ConfigurationNode;
+import org.slf4j.Logger;
+import org.spongepowered.api.block.BlockState;
+import org.spongepowered.api.block.trait.BlockTrait;
+
+import java.util.*;
+
+public class TEAction {
+
+    private String action;
+    private String targetId;
+
+    private String growthTrait = null;
+    private String idTrait = null;
+    private Map<String, TEActionReward> rewards = new HashMap<>();
+
+    private TEActionReward reward;
+
+    public void loadConfigNode(String action, ConfigurationNode node) {
+        ConfigurationNode idTraitNode = node.getNode("id-trait");
+        ConfigurationNode growthTraitNode = node.getNode("growth-trait");
+
+        if (idTraitNode.isVirtual()) {
+            this.action = action;
+            this.targetId = node.getKey().toString();
+            this.reward = new TEActionReward();
+            this.reward.loadConfigNode(node);
+            this.growthTrait = growthTraitNode.getString(null);
+            return;
+        }
+
+        String traitName = idTraitNode.getString("type");
+
+        ConfigurationNode idTraits = node.getNode(traitName);
+        rewards.clear();
+
+        if (idTraits.hasMapChildren()) {
+            idTraits.getChildrenMap().forEach((k, v) -> {
+                if (!(k instanceof String)) {
+                    return;
+                }
+
+                TEActionReward reward = new TEActionReward();
+                reward.loadConfigNode(v);
+                rewards.put(((String) k), reward);
+            });
+        }
+        this.action = action;
+        this.targetId = node.getKey().toString();
+        this.idTrait = idTraitNode.getString("type");
```

Comment:
This should be set to null by default, otherwise [isIdTraited](https://github.com/Erigitic/TotalEconomy/pull/199/files#diff-39abf54c927219e9547d29f8fed52302R85) will return true everytime.

Reply:
Due to the code above this case won't happen as the node has to be explicitly set in order for this line to be reached. See ll.50-56.
The field defaults to null and the method early-exits when the id-trait node is virtual.